ExNavigator.js

```diff
@@ -88,12 +85,14 @@ export default class ExNavigator extends React.Component {
       return null;
     }
 
-    return (
+    let navigationBar = (
       <Navigator.NavigationBar
         routeMapper={this._routeRenderer.navigationBarRouteMapper}
         style={[ExNavigatorStyles.bar, this.props.navigationBarStyle]}
       />
     );
+
+    return this.augmentNavigationBar(navigationBar);
```

Comment:
Looking at this again I think a prop would be preferable:

```js
let navigationBar = this.props.renderNavigationBar({
  routeMapper: this._routeRenderer.navigationBarRouteMapper,
  style: [ExNavigatorStyles.bar, this.props.navigationBarStyle],
});
```

And use `defaultProps` to render a basic `Navigator.NavigationBar` by default.

For the inheritance case I think we could override `defaultProps`:
```js
class CustomNavigator extends ExNavigator {
  static defaultProps = {
    ...ExNavigator.defaultProps,
    renderNavigationBar: props => <CustomNavigationBar {...props} />,
  };
}
```